src/http_request.cpp

```diff
@@ -155,7 +166,13 @@ MHD_Result http_request::build_request_args(void *cls, enum MHD_ValueKind kind,
     std::string value = ((arg_value == nullptr) ? "" : arg_value);
 
     http::base_unescaper(&value, aa->unescaper);
-    (*aa->arguments)[key] = value;
+    auto existing_iter = aa->arguments->find(key);
+    if (existing_iter != aa->arguments->end()) {
+        // Key exists, add value to collection instead of overwriting previous value.
```

Comment:
Whether the key exists or not, I think this if/else equivalent to just:

    (*aa->arguments)[key].push_back(value);